tests/plans/test_get_run_engine_metadata.py

```diff
@@ -0,0 +1,54 @@
+import re
+
+import pytest
+from bluesky import RunEngine
+
+from dodal.plans import get_run_engine_metadata
+
+
+def test_good_get_run_engine_metadata(run_engine: RunEngine):
+    test_string = "blah"
+    test_int = 5
+    test_float = 4.4
+
+    run_engine.md["test_string"] = test_string
+    run_engine.md["test_int"] = test_int
+    run_engine.md["test_float"] = test_float
+
+    def test_plan():
+        assert (yield from get_run_engine_metadata("test_string")) == test_string
+        assert (yield from get_run_engine_metadata("test_int", int)) == test_int
+        assert (yield from get_run_engine_metadata("test_float", int)) == test_float
```

Comment:
I think you miss typed here causing your test to fail
```suggestion
        assert (yield from get_run_engine_metadata("test_float", float)) == test_float
```